src/GrasshopperTeklaDrawingLink/Components/Text/FontAttributesComponent.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System.Collections.Generic;
+using System.Drawing;
+
+using Grasshopper.Kernel;
+
+using GTDrawingLink.Properties;
+using GTDrawingLink.Tools;
+using GTDrawingLink.Types;
+
+using Tekla.Structures.Drawing;
+
+
+namespace GTDrawingLink.Components.Text {
+    public class FontAttributesComponent : TeklaComponentBase {
+        public override GH_Exposure Exposure => GH_Exposure.primary;
+        protected override Bitmap Icon => Resources.Font;
+
+        public FontAttributesComponent()
+            : base(ComponentInfos.FontAttributesComponent) {
+        }
+
+        protected override void RegisterInputParams(GH_InputParamManager pManager) {
+           SetParametersAsOptional(pManager,new List<int> { pManager.AddParameter(new EnumParam<DrawingColors>(ParamInfos.DrawingColor, GH_ParamAccess.item)) });
```

Comment:
You have not utilized full power of SetParametersAsOptional.
Idea for this method is great - let's propagate it through the codebase, starting from new components like this.